### PR TITLE
Added new values PaymentSourceType

### PR DESCRIPTION
**Pull request by AleGladchteinParser**

### CODE REVIEW

File: src/test/java/com/checkout/payments/PaymentsClientImplTest.java [hunk 1]
```diff
@@ -79,6 +84,41 @@ void shouldRequestPayment() throws ExecutionException, InterruptedException {
...
+                .thenReturn(CompletableFuture.completedFuture(response));
+
+        final CompletableFuture<PaymentResponse> future = paymentsClient.requestPayment(request);
+
+        assertNotNull(future.get());
+        assertEquals(response, future.get());
+
+    }
+
+    private class CustomerSource extends AbstractRequestSource
```
Comment (Armando-Ibarra) [inline src/test/java/com/checkout/payments/PaymentsClientImplTest.java:107]: I see what you are trying to do.
Hmmm, I don't have an opinion to adding inner classes for testing.

My suggestions are:
-In Java properties goes first, then constructors.

**Review by Armando-Ibarra — APPROVED** [review]
LGTM

File: src/test/java/com/checkout/payments/PaymentsClientImplTest.java [hunk 2]
```diff
@@ -79,6 +84,41 @@ void shouldRequestPayment() throws ExecutionException, InterruptedException {
 
     }
 
+    @Test
+    void shouldRequestPayment_customerSource() throws ExecutionException, InterruptedException {
```
Comment (rsaestrela) [inline src/test/java/com/checkout/payments/PaymentsClientImplTest.java:88]: Minor detail: this should be `customSource`.

File: src/test/java/com/checkout/payments/PaymentsClientImplTest.java [hunk 3]
```diff
@@ -16,6 +20,7 @@
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import javax.swing.plaf.IconUIResource;
```
Comment (rsaestrela) [inline src/test/java/com/checkout/payments/PaymentsClientImplTest.java:23]: This import is not necessary

File: src/test/java/com/checkout/payments/PaymentsClientImplTest.java [hunk 4]
```diff
@@ -79,6 +84,41 @@ void shouldRequestPayment() throws ExecutionException, InterruptedException {
 
     }
 
+    @Test
+    void shouldRequestPayment_customerSource() throws ExecutionException, InterruptedException {
+
+        CustomerSource customerSource = new CustomerSource(10, Currency.USD);
+
+        final PaymentRequest request = mock(PaymentRequest.class);
+        request.setSource(customerSource);
```
Comment (rsaestrela) [inline src/test/java/com/checkout/payments/PaymentsClientImplTest.java:93]: Please replace this mock with a real object - something like `PaymentRequest.builder().source(customSource).build()`. I think it makes more sense in this context.

File: src/main/java/com/checkout/common/PaymentSourceType.java [hunk 5]
```diff
@@ -39,6 +39,23 @@
...
+    ALIPAY,
+    @SerializedName("paypal")
+    PAYPAL,
+    @SerializedName("multibanco")
+    MULTIBANCO,
+    @SerializedName("eps")
+    EPS,
+    @SerializedName("poli")
+    POLI,
+    @SerializedName("P24")
```
Comment (rsaestrela) [inline src/main/java/com/checkout/common/PaymentSourceType.java:55]: Serialized name is `p24`, not `P24`. Please review the list and make it 100% equal to the .NET version.

**Review by rsaestrela — CHANGES_REQUESTED** [review]

**Review by rsaestrela — APPROVED** [review]

**Pull request merged**